File: app/lib/models/guide_request.dart
class GuideRequest {
  final String fromUserId;
  final String toUserId;
  final String meetingLocation;
  final DateTime scheduledTime;
  final String duration;
  final String status;
  final int fee;
  final DateTime createdAt;

  GuideRequest({
    required this.fromUserId,
    required this.toUserId,
    required this.meetingLocation,
    required this.scheduledTime,
    required this.duration,
    required this.status,
    required this.fee,
    required this.createdAt,
  });

  factory GuideRequest.fromJson(Map<String, dynamic> json) => GuideRequest(
        fromUserId: json['from_user_id'],
        toUserId: json['to_user_id'],
        meetingLocation: json['meeting_location'],
        scheduledTime: DateTime.parse(json['scheduled_time']),
        duration: json['duration'],
        status: json['status'],
        fee: json['fee'],
        createdAt: DateTime.parse(json['created_at']),
      );

  Map<String, dynamic> toJson() => {
        'from_user_id': fromUserId,
        'to_user_id': toUserId,
        'meeting_location': meetingLocation,
        'scheduled_time': scheduledTime.toIso8601String(),
        'duration': duration,
        'status': status,
        'fee': fee,
        'created_at': createdAt.toIso8601String(),
      };
}

File: functions/entity/guide_request.js
// guide_requests 컬렉션 (오프라인 가이드 요청)
// 문서 ID: guideRequestId (자동 생성)
// {
//   from_user_id: string, // traveler
//   to_user_id: string,   // local
//   meeting_location: string,
//   scheduled_time: string, // ISO8601
//   duration: string, // 예: '3h'
//   status: 'pending' | 'accepted' | 'completed',
//   fee: number,
//   created_at: timestamp
// }

class GuideRequest {
  constructor(data) {
    this.from_user_id = data.from_user_id;
    this.to_user_id = data.to_user_id;
    this.meeting_location = data.meeting_location;
    this.scheduled_time = data.scheduled_time;
    this.duration = data.duration;
    this.status = data.status; // 'pending' | 'accepted' | 'completed'
    this.fee = data.fee;
    this.created_at = data.created_at;
  }
}

module.exports = GuideRequest; 